opm/input/eclipse/Schedule/UDQ/UDQFunction.cpp

```diff
@@ -426,58 +426,94 @@ UDQSet UDQBinaryFunction::eval(const UDQSet& lhs, const UDQSet& rhs) const
     return this->func(lhs, rhs);
 }
 
-UDQSet UDQBinaryFunction::LE(double eps, const UDQSet& lhs, const UDQSet& rhs)
+UDQSet UDQBinaryFunction::LE(const double eps, const UDQSet& lhs, const UDQSet& rhs)
 {
     auto result = lhs - rhs;
-    auto rel_diff = result / lhs;
 
-    for (std::size_t index = 0; index < result.size(); ++index) {
-        if (auto elm = result[index]; elm) {
-            if (const double abs_diff = elm.get(); abs_diff == 0) {
-                result.assign(index, 1);
-            }
-            else {
-                result.assign(index, ! (rel_diff[index].get() > eps));
-            }
+    auto is_le = [eps,
+                  scalar_lhs = is_scalar(lhs),
+                  scalar_rhs = is_scalar(rhs),
+                  &lhs, &rhs]
+        (const std::size_t i)
+    {
+        const auto x = scalar_lhs ? lhs[0].get() : lhs[i].get();
```

Comment:
i think i'd do all the extra logic outside the lambda to keep it on topic.

ie
```c++
auto is_le = [eps](const auto x, const auto y) { return ...; }
```
and then
```c++
result.assign(index, is_le(scalar_lhs ? lhs[0] : lhs[index], 
                           scalar_rhs ? rhs[0] : rhs[index]);
```
but then again these are local and not meant for reusability i guess,

Reply:
> i think i'd do all the extra logic outside the lambda to keep it on topic.

That's a valid suggestion, certainly.  It's a little annoying, in this context, that [`UDQSet::operator[]()`](https://github.com/OPM/opm-common/blob/a01e8675a661cbd5879a717e30ff8130f027dcf5/opm/input/eclipse/Schedule/UDQ/UDQSet.hpp#L458-L463) returns a `UDQScalar` which means we need to call [`UDQScalar::get()`](https://github.com/OPM/opm-common/blob/a01e8675a661cbd5879a717e30ff8130f027dcf5/opm/input/eclipse/Schedule/UDQ/UDQSet.hpp#L148-L151) to retrieve the actual value.  Then again, the lambdas could accept `UDQScalar`s directly and just call `.get()` internally...

Reply:
> right. keep it or change it, mostly a "generic" thing, current code does the job just fine as such.

I decided to switch to using a more generic formulation by extracting a helper function to run the loop.  Then the comparison functions need only care about the specifics of their individual comparison operation.